src/components/User.vue

```diff
@@ -0,0 +1,242 @@
+			<fieldset>
+				<div v-if="message !== ''" class="notecard error">
+					{{ message }}
+				</div>
+				<p v-else>
+					{{ t('registration', 'Welcome, you can create your account below.') }}
+				</p>
+
+				<div v-if="additionalHint" class="notecard success">
+					{{ additionalHint }}
+				</div>
+
+				<p v-if="!emailIsOptional || email.length > 0" class="input">
+					<input id="email"
+						type="email"
+						class="input__field"
+						name="email"
+						:value="email"
+						disabled>
+					<label for="email" class="infield">{{ t('registration', 'Email') }}></label>
+					<img id="email-icon"
+						class="input__icon"
+						:src="emailIconPath"
+						alt="">
+				</p>
+
+				<p v-if="!emailIsLogin" class="input">
+					<input id="loginname"
+						type="text"
+						name="loginname"
+						class="input__field"
+						:value="loginname"
+						:placeholder="t('registration', 'Login name')"
+						required>
+					<label for="loginname" class="infield">{{ t('registration', 'Login name') }}</label>
+					<img id="loginname-icon"
+						class="input__icon"
+						:src="authIconPath"
+						alt="">
+				</p>
+				<input v-else
+					type="hidden"
+					name="loginname"
+					:value="email">
+
+				<p v-if="showFullname" class="input">
+					<input id="fullname"
+						type="text"
+						name="fullname"
+						class="input__field"
+						:value="fullname"
+						:placeholder="t('registration', 'Full name')"
+						:required="enforceFullname">
+					<label for="fullname" class="infield">{{ t('registration', 'Full name') }}</label>
+					<img id="fullname-icon"
+						class="input__icon"
+						:src="userIconPath"
+						alt="">
+				</p>
+				<input v-else
+					type="hidden"
+					name="fullname"
+					value="">
+
+				<p v-if="showPhone" class="groupmiddle input">
+					<input id="phone"
+						type="text"
+						name="phone"
+						class="input__field"
+						:value="phone"
+						:placeholder="t('registration', 'Phone number')"
+						:required="enforcePhone">
+					<label for="phone" class="infield">{{ t('registration', 'Phone number') }}</label>
+					<img id="phone-icon"
+						class="input__icon"
+						:src="phoneIconPath"
+						alt="">
+				</p>
+				<input v-else
+					type="hidden"
+					name="phone"
+					value="">
+
+				<p class="groupbottom input">
+					<input id="password"
+						type="password"
+						class="input__field"
+						name="password"
+						:value="password"
+						:placeholder="t('registration', 'Password')"
+						required>
+					<label for="password" class="infield">{{ t('registration', 'Password') }}</label>
+					<img id="password-icon"
+						class="svg input__icon"
+						:src="passwordIconPath"
+						alt="">
+					<Button class="toggle-password"
+						type="tertiary-no-background"
+						:aria-label="isPasswordHidden ? t('registration', 'Show password') : t('registration', 'Hide password')"
+						@click.stop.prevent="togglePassword">
```

Comment:
missing keyboard accessibility
```suggestion
						@click.stop.prevent="togglePassword"
						@keydown.enter="togglePassword">
```

Reply:
But also mouse clicking doesn't toggle the visibility, because passwordInputType is not used